module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -17,13 +21,14 @@ type ApplyBatchResult<T> = Result<T, ApplyBatchError>;
 /// An implementation of the
 /// [`StateTransitionFunction`](sov_rollup_interface::stf::StateTransitionFunction)
 /// that is specifically designed to work with the module-system.
-pub struct AppTemplate<C: Context, RT, Vm, B> {
+pub struct AppTemplate<C: Context, RT, Vm, Cond, B> {
```

Comment:
If a module depends on an additional generic it leaks to the `AppTemplate` and this won't scale. Very often  additional generics can be avoided by putting them in a Box. @preston-evans98 any thoughts?

Reply:
So `Cond` is more general than just the `chain-state` module. It can be used as a general way to prove additional constraints that are external to the zk circuit, and in that sense it has the same importance as the `Vm` generic. That is the reason why I decided to introduce it there